src/desktop/apps/auction/routes.js

```diff
@@ -84,9 +84,14 @@ export async function index(req, res, next) {
       sort = "-searchable_estimate"
     }
 
-    const modal = req.originalUrl.match("/confirm-registration")
-      ? "ConfirmRegistration"
-      : null
+    let modal
+    if (req.originalUrl.match("/confirm-registration")) {
+      if (req.query.origin === "bid") {
+        modal = "ConfirmBidAndRegistration"
+      } else {
+        modal = "ConfirmRegistration"
+      }
+    }
```

Comment:
Is this logic actually meaningful? It seems like we are always showing the [`<ConfirmRegistrationModal>`](https://github.com/artsy/force/pull/4316/files#diff-16938441f4032b526aa6ac9e30307deeR34).

Reply:
We have an upcoming ticket that will add more complexity here (for showing the checkbox). I don't think it's ideal but its where the modal rendering logic lives for now (partly due to SSR issues that i don't completely understand). I'd like to refactor/move it in the future for sure.